client-guide/jmap-client-guide.mdown

```diff
@@ -11,278 +9,294 @@ A single login may have access to multiple accounts. In this guide I'm going to
 
 ## Paging in data as the interface is navigated
 
-In our example Inbox, there are 1213 threads, but so far we have only loaded in data for the first 10. As the user scrolls down, we need to page in the data for the section of the mailbox that becomes visible (and indeed, to avoid the user having to wait, it's advisable to preload a little way ahead too). This is just another call to `getMessageList`, as in the cold boot example, but with the `position` property changed to the index for the section required.
+In our example Inbox, there are 1213 threads, but so far we have only loaded in data for the first 10. As the user scrolls down, we need to page in the data for the section of the mailbox that becomes visible (and indeed, to avoid the user having to wait, it's advisable to preload a little way ahead too). This is just another call to `Email/query`, as in the cold boot example, but with the `position` property changed to the index for the section required.
 
 Similarly, if we switch mailboxes, or want to do a search, we can use the same call as well. However, remember in JMAP the same message may appear in multiple mailboxes, so to avoid downloading the same data multiple times, it's advisable to just fetch the message list without also getting the message or thread objects (except in certain situations, such as the very first request; the exact heuristics for deciding between the two can be made arbitrarily clever and complex).
 
-    [
-        [ "getMessageList", {
-            filter: {
-                inMailboxes: [ "mailbox2" ]
-            },
-            sort: [ "date desc", "id desc" ]
-            collapseThreads: true,
-            position: 11,
-            limit: 10,
-            fetchThreads: false,
-            fetchMessages: false,
-            fetchMessageProperties: null,
-            fetchSearchSnippets: false
-        }, "call1"]
-    ]
-
-After the list has been returned, a second request can be made directly to `getThreads` with the thread ids (or to `getMessages` with the message ids) to fetch the ones you are missing.
+```json
+[
+    [ "Email/query", {
+        "accountId": "u123456",
+        "filter": {
+            "inMailbox": "mailbox1"
+        },
+        "sort": [
+            { "property": "receivedAt", "isAscending": false }
+        ],
+        "collapseThreads": true,
+        "position": 10,
+        "limit": 10
+    }, "0" ]
+]
+```
+
+After the list has been returned, a second request can be made directly to `Thread/get` with the thread ids (or to `Email/get` with the message ids) to fetch the ones you are missing.
```

Comment:
We don't have the thread ids at this point; you'd have to chain it off `Email/get`.